### PR TITLE
Story #599 Edit agency wishcard on /wishcards/me

### LINKED ISSUE #599: STORY: Wishcard Edit Feature
Screenshot 2023-08-07 at 4 12 41 PM

Page: https://donate-gifts.com/wishcards/me

- [x] Add pencil icon button on each top right corner of the wish card image (same as https://donate-gifts.com/profile edit icon) 
- [x] Add modal for editing each wish card (similar to profile panel edits) 

Editable fields:
- wish item name 
- price
- child interest
- child story
- child name

- [x] Validate all fields, check not empty, max char, price limit
- [x] Test save

### PR DESCRIPTION
**Pull request by nguyenit67**
Resolve #599 Edit agency wishcard feature on /wishcards/me

### CODE REVIEW

File: src/api/wishcards.ts [hunk 1]
```diff
@@ -1,8 +1,21 @@
...
+
+router.get('/agency_me', Permissions.isAdminOrAgency, wishCardController.getAgencyWishcards);
+
+router.put(
+	'/agency_me',
+	Permissions.isAdminOrAgency,
+	Validations.updateAgencyWishcardValidationRules(),
+	Validations.validate,
+	wishCardController.putAgencyWishCardById,
+);
```
Comment (Enubia) [inline src/api/wishcards.ts:19]: we should consider renaming this, <something>_me is not clear what it means
If it's related to wish cards, then name it accordingly

**Review by Enubia — CHANGES_REQUESTED** [review]

File: src/api/wishcards.ts (hunk 1 above)
Comment (nguyenit67) [reply src/api/wishcards.ts:19]: so should '/api/wishcards/agency' works?

Comment (stacysealky) [reply src/api/wishcards.ts:19]: Yeah I agree that "me" is not very clear. How about  /wishcard/manage? It's more consistent with the action button that says "manage wish cards" too.

Comment (nguyenit67) [reply src/api/wishcards.ts:19]: so are you guys ok with '/wishcards/manage'  for the manage wishcards page route for partner user and '/api/wishcards/agency/manage' for the api route ? @stacysealky @Enubia

Comment (Enubia) [reply src/api/wishcards.ts:19]: Yes that would be ok

**Review by Enubia — APPROVED** [review]

**Pull request merged**